MovingBlocks/Terasology
# refactor: remove vr support

### Contains

Removes classes and code related to VR Support as part of the scope reduction for the 2023 revival efforts.

Removed classes:
* OpenVRInput
* OpenVRControllers
* OpenVRStereoCamera
* OpenVRProvider
* OpenVRState
* OpenVRUtils
* ControllerListener

Adjustments (removed code paths and usages) in:
* Terasology.java
* StateInGame
* WorldRendererImpl
* LocalPlayerSystem
* FirstPersonHeldItemMountPointComponent
* RenderingConfig
* Video Settings Menu
* ModuleTestingEnvironment
* IntegrationEnvironment

### How to test

Brief description of how to test / confirm this PR before merging

* Compile Terasology, verify that no build errors
* Start Terasology and CoreGameplay, verify that starts properly
* Check Video Settings Menu looks fine

### Outstanding before merging

- [x] Discuss whether and potentially decide that we indeed want to remove VR support

### Dependencies

Requires https://github.com/Terasology/CoreRendering/pull/74 and https://github.com/Terasology/ModuleTestingEnvironment/pull/77
